src/test/java/org/jabref/gui/preferences/journals/JournalAbbreviationsViewModelTabTest.java

```diff
@@ -39,32 +39,62 @@
+class JournalAbbreviationsViewModelTabTest {
 
     private JournalAbbreviationsTabViewModel viewModel;
     private Path emptyTestFile;
-    private Path testFile1Entries;
-    private Path testFile3Entries;
-    private Path testFile4Entries;
-    private Path testFile5EntriesWithDuplicate;
-    private JournalAbbreviationPreferences abbreviationPreferences;
+    private Path tempFolder;
+    private PreferencesService preferencesService;
     private final JournalAbbreviationRepository repository = JournalAbbreviationLoader.loadBuiltInRepository();
     private DialogService dialogService;
 
+    public static Stream<Arguments> provideTestFiles() {
+        return Stream.of(
+                // Mixed abbreviations
+                Arguments.of(
+                        List.of(new String[]{"testFile1Entries.csv", "Test Entry;TE" + NEWLINE + ""},
+                                new String[]{"testFile3Entries.csv", "Abbreviations;Abb;A" + NEWLINE + "Test Entry;TE" + NEWLINE + "MoreEntries;ME;M" + NEWLINE + ""},
+                                new String[]{"testFile4Entries.csv", "Abbreviations;Abb" + NEWLINE + "Test Entry;TE;T" + NEWLINE + "MoreEntries;ME;M" + NEWLINE + "Entry;E" + NEWLINE + ""},
+                                new String[]{"testFile5Entries.csv", "Abbreviations;Abb" + NEWLINE + "Test Entry;TE;T" + NEWLINE + "Test Entry;TE" + NEWLINE + "MoreEntries;ME;M" + NEWLINE + "EntryEntry;EE" + NEWLINE + ""}),
+                        List.of(
+                                List.of("Abbreviations;Abb;Abb", "Test Entry;TE;T", "MoreEntries;ME;M", "JabRefTestEntry;JTE;JTE"),
+                                List.of("EntryEntry;EE;EE", "Abbreviations;Abb;Abb", "Test Entry;TE;T", "SomeOtherEntry;SOE;SOE"))),
+
+                // No shortest unique abbreviations
+                Arguments.of(
+                        List.of(new String[]{"testFile1Entries.csv", "Test Entry;TE" + NEWLINE + ""},
+                                new String[]{"testFile3Entries.csv", "Abbreviations;Abb" + NEWLINE + "Test Entry;TE" + NEWLINE + "MoreEntries;ME" + NEWLINE + ""},
+                                new String[]{"testFile4Entries.csv", "Abbreviations;Abb" + NEWLINE + "Test Entry;TE" + NEWLINE + "MoreEntries;ME" + NEWLINE + "Entry;E" + NEWLINE + ""},
+                                new String[]{"testFile5Entries.csv", "Abbreviations;Abb" + NEWLINE + "Test Entry;TE" + NEWLINE + "Test Entry;TE" + NEWLINE + "MoreEntries;ME" + NEWLINE + "EntryEntry;EE" + NEWLINE + ""}),
+                        List.of(
+                                List.of("Abbreviations;Abb;Abb", "Test Entry;TE;TE", "MoreEntries;ME;ME", "JabRefTestEntry;JTE;JTE"),
+                                List.of("EntryEntry;EE;EE", "Abbreviations;Abb;Abb", "Test Entry;TE;TE", "SomeOtherEntry;SOE;SOE"))),
+
+                // Shortest unique abbreviations
+                Arguments.of(
+                        List.of(new String[]{"testFile1Entries.csv", "Test Entry;TE;T" + NEWLINE + ""},
```

Comment:
Can't we use `List<String>` instead of `String[]`? 

```suggestion
                        List.of(List.of("testFile1Entries.csv", "Test Entry;TE;T" + NEWLINE + ""),
```